docker-compose.yml

```diff
@@ -9,7 +9,7 @@ services:
       - ./CIHM-Swift:/home/tdr/CIHM-Swift
     networks:
       - access-platform
-    command: ["echo", "'unsure'"]
+    command: [ "perl", "./CIHM-Meta/bin/dmdtask" ]
```

Comment:
How do you set it up on the server? Does it just execute these files in a cron job? :o I'm curious~

Reply:
On production servers I set up multiple commands within multiple containers, so several can run in parallel.  You don't want something this complex for testing, but just so you can see what a production docker-compose.yml file looks like (the variables with all the passwords are of course elsewhere, in this case within /etc/canadiana/metadatabus/.env )

This is Jarlsberg, where most of the work is done:

```
version: "3.9"
# This file is distributed by puppet
services:
  dmdtask:
    image: ${mbusimage}
    init: true
    restart: unless-stopped
    command:
      - bash
      - -c
      - while :; do dmdtask ; sleep 1m ; done
    volumes:
      - /var/log/tdr:/var/log/tdr
      - /var/lock/tdr:/var/lock/tdr
      - /data/packaging/Temp/reports:/home/tdr/reports
    env_file:
      - /etc/canadiana/metadatabus/.env
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
  smelter:
    image: ${mbusimage}
    init: true
    restart: unless-stopped
    command:
      - bash
      - -c
      - while :; do smelter --maxprocs=10 --timelimit=14400 ; sleep 10m ; done
    volumes:
      - /var/log/tdr:/var/log/tdr
      - /var/lock/tdr:/var/lock/tdr
    env_file:
      - /etc/canadiana/metadatabus/.env
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
  importocr:
    image: ${mbusimage}
    init: true
    restart: unless-stopped
    command:
      - bash
      - -c
      - sleep 15 ; while :; do importocr --maxprocs=20 --timelimit=86400 ; sleep 10m ; done
    volumes:
      - /var/log/tdr:/var/log/tdr
      - /var/lock/tdr:/var/lock/tdr
    env_file:
      - /etc/canadiana/metadatabus/.env
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
  hammer2:
    image: ${mbusimage}
    init: true
    restart: unless-stopped
    command:
      - bash
      - -c
      - while :; do hammer2 --maxprocs=10 --timelimit=14400 ; sleep 10m ; done
    volumes:
      - /var/log/tdr:/var/log/tdr
      - /var/lock/tdr:/var/lock/tdr
    env_file:
      - /etc/canadiana/metadatabus/.env
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
  press2:
    image: ${mbusimage}
    init: true
    restart: unless-stopped
    command:
      - bash
      - -c
      - sleep 30 ; while :; do press2 ; sleep 10m ; done
    volumes:
      - /var/log/tdr:/var/log/tdr
      - /var/lock/tdr:/var/lock/tdr
    env_file:
      - /etc/canadiana/metadatabus/.env
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
  reposync:
    image: ${mbusimage}
    init: true
    restart: unless-stopped
    user: tdr
    command:
      - bash
      - -c
      - while :; do reposync --since=6hours ; sleep 10m ; done
    volumes:
      - /var/log/tdr:/var/log/tdr
      - /var/lock/tdr:/var/lock/tdr
    env_file:
      - /etc/canadiana/metadatabus/.env
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
```


This is Cheddar, where it only has to deal with streaming data to Solr.  No passwords in this case, so no .env file, and only the variables needed for the `solrstream` tool.

```
version: "3.9"
# This file is distributed by puppet
services:
  solrstream:
    image: ${mbimage}
    init: true
    restart: unless-stopped
    command:
      - bash
      - -c
      - while :; do solrstream ; sleep 10m ; done
    environment:
      - solrstream_couchserver=http://cheddar.uab.c7a.ca:5984
      - solrstream_couchdb=cosearch2
      - solrstream_solrserver=http://cheddar.uab.c7a.ca:8983/
      - solrstream_solrdb=cosearch2
      - solrstream_limit=30000
      - solrstream_localdocument=solrstream2.cheddar
      - solrstream_timelimit=7200
    volumes:
      - "/var/log/tdr:/var/log/tdr"
      - "/var/lock/tdr:/var/lock/tdr"
    logging:
      options:
        mode: "non-blocking"
        max-buffer-size: "4m"
        max-size: "10m"
        max-file: "3"
```

The docker-compose.yml is different on `eclipse` which is a computer in Ottawa that sends OCR related data to the Swift and CouchDB servers in Toronto.  So, different compose files for the same container doing very different things on different computers in different parts of the country :-)